dotnetv3/Support/Tests/SupportIntegrationTests.cs

```diff
@@ -0,0 +1,237 @@
+                "Example case for integration testing, please ignore.",
+                "This is an example support case.");
+        });
+    }
+
+    /// <summary>
+    /// Add an attachment set. Should return a new attachment Id.
+    /// </summary>
+    /// <returns>Async task.</returns>
+    [Fact]
+    [Order(7)]
+    public async Task VerifyAddAttachmentToSet_ShouldReturnId()
+    {
+        var fileName = _configuration["exampleFileName"];
+        // Create the file if it does not already exist.
+        if (!File.Exists(fileName))
+        {
+            await using StreamWriter sw = File.CreateText(fileName);
+            await sw.WriteLineAsync(
+                "This is a sample file for attachment to a support case.");
+        }
+
+        await using var ms = new MemoryStream(await File.ReadAllBytesAsync(fileName));
+
+        var attachmentSetId = await _supportWrapper.AddAttachmentToSet(
+            ms,
+            fileName);
+        _attachmentSetId = attachmentSetId;
+
+        Assert.False(string.IsNullOrWhiteSpace(attachmentSetId));
+    }
+
+    /// <summary>
+    /// Add a communication. Should return true.
+    /// </summary>
+    /// <returns>Async task.</returns>
+    [Fact]
+    [Order(8)]
+    public async Task VerifyAddCommunicationToCase_ShouldReturnId()
+    {
+        var result = await _supportWrapper.AddCommunicationToCase(
+            _caseId!,
+            "This is an example communication added to a support case.",
+            _attachmentSetId);
+
+        Assert.True(result);
+    }
+
+    /// <summary>
+    /// Describe the communications of a case. Should return a list that is not empty.
+    /// </summary>
+    /// <returns>Async task.</returns>
+    [Fact]
+    [Order(9)]
+    public async Task VerifyCommunicationForCase_ShouldNotBeEmpty()
+    {
+        var communications = await _supportWrapper.DescribeCommunications(_caseId!);
+        _attachmentSetId = communications.FirstOrDefault()?.AttachmentSet.FirstOrDefault()?
+            .AttachmentId;
+        Assert.NotEmpty(communications);
+    }
+
+    /// <summary>
+    /// Describe the attachment set. Should return an attachment with a filename that is not empty.
+    /// </summary>
+    /// <returns>Async task.</returns>
+    [Fact]
+    [Order(10)]
+    public async Task VerifyDescribeAttachment_ShouldNotBeEmpty()
+    {
+        var attachment = await _supportWrapper.DescribeAttachment(_attachmentSetId!);
+        Assert.NotEmpty(attachment.FileName);
+    }
+
+    /// <summary>
+    /// Describe cases in the support account. Should return a list that is not empty.
+    /// </summary>
+    /// <returns>Async task.</returns>
+    [Fact]
+    [Order(11)]
+    public async Task VerifyDescribeCases_ShouldNotBeEmpty()
+    {
+        // Describe the cases. In case it is empty, try again and allow time for the new case to appear.
+        List<CaseDetails> todayCases = null!;
+        while (todayCases == null || todayCases.Count == 0)
```

Comment:
Integration test *could* hang forever on this. Maybe include a max retries?

Reply:
Good point, updated to limit the number of retries.